docs/notebooks/using_mclip.md

```diff
@@ -0,0 +1,249 @@
+
+```python
+import finetuner
+
+finetuner.login()
+run = finetuner.get_run(run.name)
+```
+
+You can continue monitoring the run by checking the status - `finetuner.run.Run.status()` or the logs `finetuner.run.Run.logs()`.
+<!-- #endregion -->
+
+<!-- #region -->
+## Evaluating
+Once the run is finished, the metrics calculated by the {class}`~finetuner.callback.EvaluationCallback` are plotted using the {class}`~finetuner.callback.WandBLogger` callback. These plots can be accessed using the link provided in the logs once finetuning starts:
+
+```bash
+           INFO     Finetuning ... 
+wandb: Currently logged in as: anony-mouse-448424. Use `wandb login --relogin` to force relogin
+wandb: Tracking run with wandb version 0.13.5
+wandb: Run data is saved locally in <path-to-file>
+wandb: Run `wandb offline` to turn off syncing.
+wandb: Syncing run ancient-galaxy-2
+wandb:  View project at <link-to-project>
+wandb:  View run at <link-to-run>
+
+```
+
+The generated plots should look like this:
+
+![WandB-mclip](images/WandB-mclip.png)
+
+<!-- #endregion -->
+
+## Saving
+
+After the run has finished successfully, you can download the tuned model on your local machine:
+
+```python
+artifact = run.save_artifact('mclip-model')
+```
+
+## Inference
+
+Now you saved the `artifact` into your host machine,
+let's use the fine-tuned model to encode a new `Document`:
+
+```python
+text_da = DocumentArray([Document(text='setwas Text zum Codieren')])
+image_da = DocumentArray([Document(uri='https://upload.wikimedia.org/wikipedia/commons/4/4e/Single_apple.png')])
+
+mclip_text_encoder = finetuner.get_model(artifact=artifact, select_model='clip-text')
+mclip_image_encoder = finetuner.get_model(artifact=artifact, select_model='clip-vision')
+
+finetuner.encode(model=mclip_text_encoder, data=text_da)
+finetuner.encode(model=mclip_image_encoder, data=image_da)
+
+print(text_da.embeddings.shape)
+print(image_da.embeddings.shape)
+```
+
+<!-- #region -->
+```bash
+(1, 512)
+(1, 512)
+```
+
+```{admonition} what is select_model?
+When fine-tuning CLIP, we are fine-tuning the CLIPVisionEncoder and CLIPTextEncoder in parallel.
+The artifact contains two models: `clip-vision` and `clip-text`.
+The parameter `select_model` tells finetuner which model to use for inference, in the above example,
+we use `clip-text` to encode a Document with text content.
+```
+
+```{admonition} Inference with ONNX
+In case you set `to_onnx=True` when calling `finetuner.fit` function,
+please use `model = finetuner.get_model(artifact, is_onnx=True)`
+```
+<!-- #endregion -->
+
+## Before and After
+We can directly compare the results of our fine-tuned model with an untrained multilingual clip model by displaying the matches each model has for the same query, while the difference between the results of the two models are quite similar for some queries, the examples below clearly show that finetuning increses the quality of the search results:
```

Comment:
"while the difference [...] are quite similar" I think here is something wrong in the sentence?

Reply:
differences*

Reply:
yes but also "the difference are similar" makes no sense

Reply:
corrected to differences